setup/security/authentication/ssl.md

```diff
@@ -29,66 +29,140 @@ use an external CA, to minimize browser configuration, but it's not necessary
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passout pass:$DECK_KEY_PASSWORD` (place before `4096`)*
+
+1. Generate a certificate signing request for Deck. Specify `localhost` or Deck's eventual fully-qualified domain name (FQDN) as the Common Name (CN).  This will prompt for the pass phrase for `deck.key`.
+
+    ```bash
+    openssl req -new -key deck.key -out deck.csr
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passin pass:$DECK_KEY_PASSWORD`*
+
+1. Use the CA to sign the server's request and create the Deck server certificate (in `pem` format). If using an external CA, they will do this for you.  This will prompt for the pass phrase used to encrypt `ca.key`.
+
+    ```bash
+    openssl x509 -req -days 365 -in deck.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out deck.crt
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passin pass:$CA_KEY_PASSWORD`*
+
+1. Create a server key for Gate. This will prompt for a pass phrase to encrypt the key. Keep this file safe!
+
+    ```bash
+    openssl genrsa -des3 -out gate.key 4096
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passout pass:$GATE_KEY_PASSWORD` (place before `4096`)*
+
+1. Generate a certificate signing request for Gate. Specify `localhost` or Gate's eventual fully-qualified domain name (FQDN) as the Common Name (CN).  This will prompt for the pass phrase for `gate.key`.
+
+    ```bash
+    openssl req -new -key gate.key -out gate.csr
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passin pass:$GATE_KEY_PASSWORD`*
+
+1. Use the CA to sign the server's request and create the Gate server certificate (in `pem` format). If using an external CA, they will do this for you.  This will prompt for the pass phrase used to encrypt `ca.key`.
+
+    ```bash
+    openssl x509 -req -days 365 -in gate.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out gate.crt
+    ```
+    *Optional flag to avoid pass phrase prompt: `-passin pass:$CA_KEY_PASSWORD`*
+
+1. Convert the `pem` format Gate server certificate into a PKCS12 (`p12`) file, which is importable into a Java Keystore (JKS).  This will first prompt for the pass phrase used to encrypt `gate.key`, and then for an import/export password to use to encrypt the `p12` file.
+
+    ```bash
+    openssl pkcs12 -export -clcerts -in gate.crt -inkey gate.key -out gate.p12 -name gate
+    ```
+    * *Optional flag to avoid gate.key pass phrase prompt: `-passin pass:$GATE_KEY_PASSWORD`*
+    * *Optional flag to avoid export password prompt: `-password pass:$GATE_EXPORT_PASSWORD`*
+
+    This creates a p12 keystore file with your certificate imported under the alias "gate".
+
+1. Create a new Java Keystore (JKS) containing your `p12`-formatted Gate server certificate.  Because Gate assumes that the keystore password and the password for the key in the keystore are the same, we must provide both via the command line.  This will prompt for the import/export password used to encrypt the `p12` file.
```

Comment:
I think it logically makes more sense to create a JKS containing our Gate server cert and add the CA than to create the JKS with the CA and add the cert.